### ISSUE TITLE
\ref command does not insert title of referenced page (works on 1.8.14, broken in 1.9.5)

### ISSUE THREAD

**Issue by KasperHesse** [doxygen/doxygen#9719]
**Describe the bug**
Under Doxygen 1.8.14, using the `\ref` command to reference a named page, the inserted link would take on the title of the page in question. Under Doxygen 1.9.5, this no longer works, and the link is named after the page's `<name>` and not the title.

**Expected behavior**
The link should reflect the title of the page.

**Screenshots**
Expected result (from Doxygen 1.8.14)
![image](https://user-images.githubusercontent.com/47347446/207039609-10a88aa2-d0b2-4392-b73d-4e4a4392ef5c.png)

Actual result (from Doxygen 1.9.5)
![image](https://user-images.githubusercontent.com/47347446/207039643-0e124e57-96b9-4ea9-bfb7-58d906c9671f.png)

**To Reproduce**
```bash
mkdir doxy && cd doxy
echo "@mainpage The main page\nThis is the main page referencing \\ref pPage1 and @ref pPage2" >> main.md
echo "@page pPage1 Page number one\nThis is page number one" >> p1.md
echo "\\page pPage2 The second page\nThis is the second page" >> p2.md
doxygen .
```

**Version**
1.9.5 vs 1.8.14. 
lsb_release:
``` 
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: Rocky
Description:    Rocky Linux release 8.7 (Green Obsidian)
Release:        8.7
Codename:       GreenObsidian
```

**Comment by albert-github** [comment]
We see that the translation of the `@page is done as follows:
```
======== Markdown =========
---- input -------
@page pPage1 Page number one
This is page number one

---- output -----
@page md_p1 Page number one\ilinebr @anchor pPage1
This is page number one

=========
```

The extra anchor command was necessary in doxygen but probably we should add an extra option to the `@anchor` containing the label like:
```
@anchor{<label>}
```
which would in the given case give as resulting markdown output:
```
@page md_p1 Page number one\ilinebr @anchor{Page number one} pPage1
 ```

**Comment by WilliamHeuser** [comment]
Just to be clear you have added the anchor command from version 1.8.14 to 1.9.5, and this is the reason the page titles are no longer being displayed? 

If so, does this mean that it is a bug that will be fixed in future versions? or has the functionality changed permanently?

**Comment by albert-github** [comment]
As indicated in the "labels" it is in my opinion a bug, I have the intention to create a proposed pull request for this so that in future the line is automatically  corrected (see my suggestion at https://github.com/doxygen/doxygen/issues/9719#issuecomment-1346432205).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9728

**Comment by dhowland** [comment]
I recently encountered this bug as well.

**Comment by albert-github** [comment]
An adjusted version of the code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
